### PR TITLE
added hw-13-1

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/model/Bucket.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package mate.academy.internetshop.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Bucket {
+    private static long idGenerator = 0;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/model/Bucket.java:7]: better create separate class and method here and generate id 
the same in all classes

File: src/main/java/mate/academy/internetshop/service/BucketService.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    Bucket update(Bucket bucket);
+
+    void delete(Long id);
+
+    Bucket addItem(Long bucketId, Long itemId);
+
+    Bucket clear(Long bucketId);
+
+    List getAllItems(Long bucketId);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/BucketService.java]:
```suggestion
    List<Item> getAllItems(Long bucketId);
```

File: src/main/java/mate/academy/internetshop/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    User create(User user);
+
+    User get(Long id);
+
+    User update(User user);
+
+    void delete(Long id);
+
+    List getOrders(Long userId);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/UserService.java]:
```suggestion
    List<Order> getOrders(Long userId);
```

File: src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        Item item = itemDao.get(itemId);
+        bucket.getItems().add(item);
+        return bucketDao.update(bucket);
+    }
+
+    @Override
+    public Bucket clear(Long bucketId) {
+        Bucket bucket = bucketDao.get(bucketId);
+        bucket.getItems().clear();
+        return bucket;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java]:
```suggestion
        return update(bucket);
```

File: src/main/java/mate/academy/internetshop/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    @Override
+    public void delete(Long id) {
+        orderDao.delete(id);
+    }
+
+    @Override
+    public Order completeOrder(List items, Long userId) {
+        List<Item> newItems = new ArrayList<>(items);
+        Order order = new Order(userId, newItems);
+        orderDao.create(order);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/impl/OrderServiceImpl.java:50]: add: userDao.get(userId).getOrders().add(order)

File: src/main/java/mate/academy/internetshop/model/Bucket.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/model/Bucket.java:7]: For example, 1 is an unique number for Bucket and 1 is an unique number for User so I have decided to make in this way. Bogdan has said it's normal too.

File: src/main/java/mate/academy/internetshop/service/BucketService.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/service/BucketService.java]: OK

File: src/main/java/mate/academy/internetshop/service/UserService.java (hunk 3 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/service/UserService.java]: OK

File: src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java (hunk 4 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java]: OK

File: src/main/java/mate/academy/internetshop/service/impl/OrderServiceImpl.java (hunk 5 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/service/impl/OrderServiceImpl.java:50]: OK

File: src/main/java/mate/academy/internetshop/model/Bucket.java (hunk 1 above)
Comment (Sofasmile) [reply src/main/java/mate/academy/internetshop/model/Bucket.java:7]: ok

**Review by Sofasmile — APPROVED** [review]
LGTM

**Pull request merged**